server/src/lib/vnc.rs

```diff
@@ -0,0 +1,121 @@
+use async_trait::async_trait;
+use propolis::common::GuestAddr;
+use propolis::dispatch::AsyncCtx;
+use propolis::hw::qemu::ramfb::Config;
+use rfb::encodings::RawEncoding;
+use rfb::rfb::{FramebufferUpdate, Rectangle};
+use rfb::server::Server;
+use slog::{debug, info, Logger};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+#[derive(Debug)]
+pub struct RamFb {
+    addr: u64,
+    width: u32,
+    height: u32,
+}
+
+impl RamFb {
+    pub fn new(addr: u64, width: u32, height: u32) -> Self {
+        Self { addr, width, height }
+    }
+}
+
+enum Framebuffer {
+    Uninitialized,
+    Initialized(RamFb),
+}
+
+#[derive(Clone)]
+pub struct PropolisVncServer {
```

Comment:
I just called it that because there's another type called `rfb::VncServer` and I found it confusing. I would prefer to name it not just `VncServer` but am down to rename it otherwise.